### DISCUSSION TITLE
STM32H747XI support in boards.txt?

### DISCUSSION THREAD

**Discussion by radiohound** [stm32duino/Arduino_Core_STM32#1588] [Variants] [answered]
Hello, I was wondering if there was a reason that XI chips like the H747XI are not included or generated in the boards.txt file? It is already in the variants list.  I tried adding the following to my boards.txt, located in my Arduino15\packages\STMicroelectronics\hardware\stm32\2.1.0 folder. but it does not get picked up in my "board part number" dropdown for genericH7 boards. 

<# Generic H747XITx
GenH7.menu.pnum.GENERIC_H747XITX=Generic H747XITx
GenH7.menu.pnum.GENERIC_H747XITX.upload.maximum_size=2097152
GenH7.menu.pnum.GENERIC_H747XITX.upload.maximum_data_size=524288
GenH7.menu.pnum.GENERIC_H747XITX.build.board=GENERIC_H747XITX
GenH7.menu.pnum.GENERIC_H747XITX.build.product_line=STM32H747xx
GenH7.menu.pnum.GENERIC_H747XITX.build.variant=STM32H7xx/H742X(G-I)H_H743X(G-I)H_H745X(G-I)H_H747X(G-I)H_H750XBH_H753XIH_H755XIH_H757XIH>

**Comment by fpistm** [comment, marked as answer]
Hi @radiohound 
All MCU have a default variant generated, anyway it requires to add manually the linker script and declare it in the boards.txt.
https://github.com/stm32duino/wiki/wiki/Add-a-new-variant-%28board%29

If you don't see it there is probably an issue in your declaration or maybe you didn't restart the Arduino IDE.
